### PR TITLE
Upgrade testing libraries

### PR DESCRIPTION
**Pull request by blairconrad**

### CODE REVIEW

File: tests/Directory.Build.props [hunk 1]
```diff
@@ -14,11 +14,11 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.9.0" />
-    <PackageReference Include="xunit.core" Version="2.4.0" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.0" />
-    <PackageReference Include="XunitXml.TestLogger" Version="2.0.0" />
-    <PackageReference Include="FluentAssertions" Version="4.19.2" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.2.0" />
```
Comment (blairconrad) [inline tests/Directory.Build.props]: There are actually some newer versions available (up to 16.4.0), but not all are compatible with all our TFMs. This is the highest common version. Would you rather fork by TFM?

Comment (thomaslevesque) [reply tests/Directory.Build.props]: No, it's fine, I think.
At some point we're going to have to drop older TFMs, if even Microsoft doesn't support them...

File: tests/FakeItEasy.Tests.Approval/ApprovedApi/FakeItEasy/net40.approved.txt [hunk 2]
```diff
@@ -344,10 +344,8 @@ namespace FakeItEasy.Configuration
 {
     public class ArgumentCollection : System.Collections.Generic.IEnumerable<object?>, System.Collections.IEnumerable
     {
-        [System.Runtime.CompilerServices.IndexerName("Item")]
-        public System.Collections.Generic.IEnumerable<string> Item { get; }
-        [System.Runtime.CompilerServices.IndexerName("Item")]
-        public int Item { get; }
+        public System.Collections.Generic.IEnumerable<string> ArgumentNames { get; }
+        public int Count { get; }
```
Comment (thomaslevesque) [inline tests/FakeItEasy.Tests.Approval/ApprovedApi/FakeItEasy/net40.approved.txt]: Where did this change come from? It doesn't seem related to the testing libraries...

Comment (thomaslevesque) [reply tests/FakeItEasy.Tests.Approval/ApprovedApi/FakeItEasy/net40.approved.txt]: Doh! Somehow I missed that change. No, you can leave it here.

**Comment by thomaslevesque** [comment]
Thanks @blairconrad!

**Comment by blairconrad** [comment]
Thank you, @thomaslevesque. Sorry I didn't call out the PublicApiGenerator upgrade.
